### PR TITLE
Clarified that slices apply to arbitrary-precision values

### LINKED ISSUE #1015: Slicing arbitrary-width integers
p4c compiler allows the statement `bit<4> x = 16[3:0]`. This suggests variable-width integers should support slices, but the spec does not list bit slices among the supported operations for arbitrary-width integers.

Based on the p4lang/p4c#3058, the spec should add this operation to section 8.7.

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #1015

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -3654,6 +3654,30 @@ expressions of type `int` must be compile-time known values.  The type
   The right operand must be either an unsigned constant of type `bit<S>` or a non-negative integer compile-time known value.
   The expression `a << b` is equal to $a \times 2^b$ while `a >> b`
   is equal to $\lfloor{a / 2^b}\rfloor$.
+- Bit slices, denoted by `[L:R]`, where `H` and `L` must be expressions that evaluate to
```
Comment (apinski-cavium) [inline p4-16/spec/P4-16-spec.mdk:3657]: Looks like a copy and pasto, `[L:R]` should be `[H:L]`

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -3506,8 +3506,8 @@ Bit-strings also support the following operations:
   Shifting by an amount greater than or equal to the width of the input
   produces a result where all bits are zero.
 - Extraction of a set of contiguous bits, also known as a slice,
-  denoted by `[L:R]`, where `L` and `R` must be expressions that evaluate to
-  non-negative compile-time known values, and `L >= R`. The types of `L` and `R`
+  denoted by `[H:K]`, where `H` and `L` must be expressions that evaluate to
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk:3509]: typo

**Review by jnfoster — APPROVED** [review]
LGTM, modulo typo.

**Review by jafingerhut — APPROVED** [review]
Except for the one typo that Mihai found and commented on, and should be fixed, these changes look good to me.

**Pull request merged**